build/php/tests/Triniti/Tests/Schemas/Sys/RedirectIdTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+declare(strict_types=1);
+
+namespace Triniti\Tests\Schemas\Sys;
+
+use PHPUnit\Framework\TestCase;
+use Triniti\Schemas\Sys\RedirectId;
+use Gdbots\Pbj\Exception\InvalidArgumentException;
+
+final class RedirectIdTest extends TestCase
+{
+    public function testEmptyUri(): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+
+        $uri = '';
+        RedirectId::fromUri($uri);
+    }
+
+    public function testFromUri(): void
+    {
+        $uri = '/unicode/fake/partial/url';
```

Comment:
@hunyoboy these inputs/tests must match exactly the inputs/tests from js to ensure they both produce the same values